sql-plugin/src/main/spark341db/scala/org/apache/spark/sql/execution/rapids/shims/PartitionedFileUtilShim.scala

```diff
@@ -0,0 +1,42 @@
+/*
```

Comment:
In order to avoid confusion. I would create a trait to extract the existing methods in `com.nvidia.spark.rapids.shims.PartitionedFileUtilsShim` then have one implementation in 340-351 shims and one in 400 shim. With this change as is, we will cause confusion with `PartitionFileUtilsShim` in com package.

Reply:
I'm happy to go along, but I'm not sure I've understood the advantage of making that change.  
The code at the call site remains unchanged (including the object name).  So the only advantage I can think of is that the trait formalizes the expected interface?

Reply:
Made the change. Is this better?

Reply:
As discussed offline. I was suggesting to keep the file in the com package and make it implement a trait that can be implemented differently for 400 shim.

Reply:
Ok, I've spoken too soon.  Combining these is becoming a mess:  `PartitionedFileUtil` wasn't available till 3.5.  My guess is that the Databricks shim is similarly borked.

I'm trying to sort this out now.

Reply:
This should be resolved now. Waiting on the build.